launch/launch/launch_service.py

```diff
@@ -285,6 +286,12 @@ def exception_handler(loop, context):
 
             # Setup custom signal hanlders for SIGINT, SIGQUIT, and SIGTERM.
             def _on_sigint(signum, frame):
+                # Ignore additional interrupts until we finish processing this one.
+                prev_handler = signal.signal(signal.SIGINT, signal.SIG_IGN)
+                if prev_handler is signal.SIG_IGN:
+                    # This function has been called re-entrantly.
+                    return
```

Comment:
Is this expected to occur and be ok? If yes and then no, should it raise an exception?

Is this the case where a signal is received between 288 and 289?

Reply:
I am anticipating a race condition between 288 and 290, yeah. I haven't read anything to think that it's not possible.

I don't think that raising there is appropriate because the situation that caused #91 will then cause an exception, where I think that if we are already interrupting then there is no cause for concern. We print the message if we receive sigints later on but that's more for the information of the user from my understanding.